src/install.rs

```diff
@@ -243,17 +243,23 @@ pub fn cinstall(ws: &Workspace, packages: &[CPackage]) -> anyhow::Result<()> {
                     lib.install(capi_config, shared_lib, &install_path_lib)?;
                 }
                 LibType::Windows => {
-                    let install_path_bin = append_to_destdir(destdir.as_deref(), &paths.bindir);
-                    create_dir_all(&install_path_bin)?;
-
                     let lib_name = shared_lib.file_name().unwrap();
-                    copy(shared_lib, install_path_bin.join(lib_name))?;
-                    let impl_lib = build_targets.impl_lib.as_ref().unwrap();
-                    let impl_lib_name = impl_lib.file_name().unwrap();
-                    copy(impl_lib, install_path_lib.join(impl_lib_name))?;
-                    let def = build_targets.def.as_ref().unwrap();
-                    let def_name = def.file_name().unwrap();
-                    copy(def, install_path_lib.join(def_name))?;
+
+                    if capi_config.library.install_subdir.is_none() {
+                        let install_path_bin = append_to_destdir(destdir.as_deref(), &paths.bindir);
+                        create_dir_all(&install_path_bin)?;
+
+                        copy(shared_lib, install_path_bin.join(lib_name))?;
+                        let impl_lib = build_targets.impl_lib.as_ref().unwrap();
+                        let impl_lib_name = impl_lib.file_name().unwrap();
+                        copy(impl_lib, install_path_lib.join(impl_lib_name))?;
+                        let def = build_targets.def.as_ref().unwrap();
+                        let def_name = def.file_name().unwrap();
+                        copy(def, install_path_lib.join(def_name))?;
+                    } else {
+                        // We assume they are plugins, install them in the custom libdir path
+                        copy(shared_lib, install_path_lib.join(lib_name))?;
```

Comment:
If you want to emulate Meson behaviour, you will want to copy the implib and the def file to the same location.

Basically, the only thing that would change if the install path is custom, is that shared_lib will be installed to `install_path_lib` instead of `install_path_bin`

Reply:
Ah, even simpler then :)